src/main/scala/io/iohk/ethereum/vm/ProgramState.scala

```diff
@@ -4,56 +4,63 @@ import akka.util.ByteString
 import io.iohk.ethereum.domain.{Address, TxLogEntry, UInt256}
 
 object ProgramState {
-  def apply[W <: WorldStateProxy[W, S], S <: Storage[S]](context: ProgramContext[W, S]): ProgramState[W, S] =
+  def apply[W <: WorldStateProxy[W, S], S <: Storage[S]](
+      vm: VM[W, S],
+      context: ProgramContext[W, S],
+      env: ExecEnv): ProgramState[W, S] = {
+
     ProgramState(
-      context = context,
-      gas = context.startGas,
+      vm = vm,
```

Comment:
why do we need vm in ProgramState?

Reply:
Because `VM` is now type-parameterised (it's just to avoid type parameterising every single method in it).